1_fetch/src/get_sb_data.R

```diff
@@ -258,11 +258,31 @@ prep_feature_vars <- function(sb_var_data, sites, retain_vars) {
     select(COMID, label, avg_annual) %>%
     pivot_wider(names_from = "label", values_from = "avg_annual")
   
+  #determine dominant physiographic region
+  phys_region <- data %>%
+    select(COMID, contains("_PHYSIO_")) %>%
+    select(-contains("AREA")) %>%
+    group_by(COMID) %>%
+    summarise_all(mean) %>%
+    pivot_longer(!COMID, names_to = "name", values_to = "value") %>%
+    mutate(unit = str_sub(name, 1, 3), 
+           region = str_sub(name, 12)) %>%
+    group_by(COMID, unit) %>%
+    summarise(dom_phys_reg_ind = which.max(value), .groups = "drop") %>%
+    mutate(dom_phys_reg = if_else(dom_phys_reg_ind < 3, 
```

Comment:
Editing my code to handle this in a more transparent way. You should see in upcoming Commit. 

There are 25 regions, but they start at 0 (no region), skip 2 for some reason... so this was to correct for that. Now using `which.max()` in a better way to return the full row (with region column) instead of just the index (which were off my 1 for regions 0 and 1).

Reply:
https://www.sciencebase.gov/catalog/file/get/57dafd3ae4b090824ffc32f1?f=__disk__9b%2Fe6%2F04%2F9be604f55425b2691158260123a914cd0efae0da&transform=1&community=National+Water-Quality+Assessment+-+Hydrologic+Systems+Team+%28NAWQA+-+HST%29&allowOpen=true